spaceship/lib/spaceship/two_step_or_factor_client.rb

```diff
@@ -131,18 +131,33 @@ def handle_two_factor(response, depth = 0)
         puts("Environment variable `SPACESHIP_2FA_SMS_DEFAULT_PHONE_NUMBER` is set, automatically requesting 2FA token via SMS to that number")
```

Comment:
This whole `if/elsif/else` block feels ready for a refactoring now that we have 4 cases: ENV var, sms already sent, sms to be sent after choosing phone, and pushed code. Probably a future PR though to not make this one more complex than it already is?

Reply:
100%, and I have a partial refactor saved locally to better handle the expanded cases and loosen the coupling between methods (e.g. splitting up `request_two_factor_code_from_phone` which tightly couples requesting a new code with prompting the user for a 2fa code and building the payload). Definitely better to include as a separate PR though to minimize the potential for regressions.